### PR TITLE
Optimize gamemodes.txt

### LINKED ISSUE #150: Isn't "survivor_max_incapacitated_count" 0 redundant in Death's Door?
If the mutation already has `SurvivorMaxIncapacitatedCount = 0` as vscript, which has priority, what's the point of setting it as a convar in gamemodes.txt?

### PR DESCRIPTION
**Pull request by alexiscoutinho**
This PR aims to optimize **gamemodes.txt** mainly by cleaning up unnecessary cvars and optimizing the formatting. Inspired by/resolves #150.

- Fixes 2 occurences of a cvar typo;
- Cleans up 9 occurences of cvars overridden by vscript;
- Cleans up 14 occurences of inheritable cvars;
- Removes 2 deprecated? cvars;
- Cleans up 2 occurences of an inheritable flag;
- Removes 1 cvar with the default value;
- Removes 4 cvars with default values and 2 inheritable ones, that consequently kinda break completeness/symmetry.

The PR also trims trailing whitespace, cleans up 3 bad newlines, fixes indentation and removes unnecessary comments (and adds 2).

### CODE REVIEW

File: root/scripts/gamemodes.txt [hunk 1]
```diff
@@ -106,7 +106,7 @@
 			// get set on 360 and dedicated for 360
 			"360_z_mega_mob_size" 25
 			"360_z_mob_spawn_max_size" 18
-			"360_z_versus_wandering_density" 0.026
+			"360_versus_wandering_zombie_density" 0.026
```
Comment (alexiscoutinho) [inline root/scripts/gamemodes.txt]: Assuming that the desired cvar is not `z_wandering_density`, which even has a farther default `0.03` as opposed to `0.027`; and that the real 360 cvar is not incorrect, as in the typo, thus following the simple prepending of `360_`.

**Review by shqke — COMMENTED** [review]
It should be absolutely encouraged to always explicitly enclose **all** tokens with double quotes (values as well), to provide token isolation and avoid confusion for users and/or any KV parser implementation.

TABs should be fine though.

**Comment by alexiscoutinho** [comment]
How would there be confusion? Any half decent user/KV parser knows that tokens are separated by whitespace unless that whitespace is enclosed by quotation marks. Quotation marks are still used where necessary in the optimized file... Since this file will hardly ever get modified in the future, I think it is a good candidate to be slimmed down. It's not encouraging this behavior everywhere else...

I don't understand what you meant with TABs. Do you want key/values to be separated with TABs if quotation marks aren't to be used?

**Comment by shqke** [comment]
In which world would you call removal of double quotes serving as optimization?

If you may need to use enquoted tokens, you should be consistent with them across the whole file to eliminate the possibility of a problem.
Consistency also assists with readability, and doing so does help to avoid the confusion for not "half decent" users and KV parsers.

If it's not the rule of the thumb that Valve would pursue, then "don't touch whats not broken" would be very well applicable here.

**Comment by alexiscoutinho** [comment]
> In which world would you call removal of double quotes serving as optimization?

It is a file size optimization and, in my opinion, a visibility/readability optimization.
> If you may need to use enquoted tokens, you should be consistent with them across the whole file to eliminate the possibility of a problem.

In a general case I could agree, but I put special attention in this file to not mistype **any** character (I even removed 2 spaces hidden before tabs). There are only 2 instances where double quotes were needed and I considered the file as a 'final' version when doing the "extensive formatting optimization".
> Consistency also assists with readability,

Yes, but is it worth adding hundreds of double quotes just because of 2 instances?
> does help to avoid the confusion for not "half decent" users and KV parsers.

Humm, should we really expect this file to be read by novices? And wouldn't they know better/prefer the syntax they use in the console, with minimum quotes?
> If it's not the rule of the thumb that Valve would pursue, then "don't touch whats not broken" would be very well applicable here.

I don't think it's that applicable here because it's not too difficult to check it wasn't recently broken. Furthermore, the KV parser in question must be a code done decades ago by Valve and definitely not 'fragile'. This is the kind of simple code that students do in their first year of coding.

There is a compromise I'm willing to take though: use double quotes on all or some flag values, i.e. not the cvars and not the key strings themselves. What do you think?

**Comment by shqke** [comment]
I've provided you with the grounds for an objectively better change in formatting, **if it were my files** to decide for.

With your opinion at places being subjective in the scope of this repository, it is not up to you to pick the formatting without argument of a higher severity, or deciding the difficulty of comparing differences. We value Valve's time, and so the changes in this repository are aimed to be as minimal as possible. If there is a larger-traffic opportunity for such changes, you'll be notified and you'll be welcome to try again.

As a compromise, you can revert commit a182b39c with rebase, so Valve don't have to validate something that never needed a change.

**Comment by alexiscoutinho** [comment]
> it is not up to you to pick the formatting without argument of a higher severity, or deciding the difficulty of comparing differences.

Yea, I've given you a lot of leeway to say this after
> There is a compromise I'm willing to take though:

:|

---
> We value Valve's time, and so the changes in this repository are aimed to be as minimal as possible.

I didn't think about this and it does change the picture a lot. I wonder if you were thinking this from the start but withheld it (to be nicer) and only made it explicit because I forced your hand... Sorry.
Well, I can't say I'm surprised with all this since I even split the formatting changes in two commits, knowing the second one would be controversial.

It would be interesting to try a rebase on a live branch (already in PR) since I still haven't done such one, but I'll instead make a new commit because I want to keep a record of that idea.

**Comment by alexiscoutinho** [comment]
I've rebased the branch to remove the "Extensive formatting optimization" from the commit history. But a question still remains, if `versus_tank_flow_team_variation` and `z_special_spawn_interval` are indeed deprecated.

**Comment by Nesciuse** [comment]
Pointless question really. Why should someone have to bother guessing if they do anything? There is no harm keeping them as they are. Really dont see the point of doing these changes other than the cvar typo fixes. Everything else is just nitpicking.

**Comment by alexiscoutinho** [comment]
Well... :|

But I really appreciate your kind and sincere response. I understand the current circumstances barring this kind of PR. About the cvar typos, your views actually take me by surprise because, afaik, those changes are in a sense the most unnecessary ones as these recent game updates, including the TLS update, don't affect the 360 version... Or am I mistaken? If not, then I would prefer to reraise this PR in the future, when the underlying circumstances have advanced.

**Comment by Rayman1103** [comment]
Correct, the Cvar typos only affect the Xbox 360 version and won't affect PC.

I was mainly using those typos as an example, since they would affect gameplay and therefore be eligible to be accepted.

**Pull request closed without merging**